pkg/infra/iac3/templates/aws/ecs_task_definition/container_definitions.ts.tmpl

```diff
@@ -0,0 +1,103 @@
+[
+    {{- range $i, $cd := . }}
+    {
+        {{- if $cd.Command }}
+        command: [
+        {{ range $i, $cmd := $cd.Command }}
+            "{{ $cmd }}",
+        {{- end }}
+        ],
+        {{- end }}
+        {{- if $cd.Cpu }}
+        cpu: {{ $cd.Cpu }},
+        {{- end }}
+        {{- if $cd.Environment }}
+        environment: [
+            {{- range $key, $value := $cd.Environment }}
+            {
+                name: "{{ $value.Name }}",
+                value: {{ modelCase $value.Value }},
+            },
+            {{- end }}
+        ],
+        {{- end }}
+        {{- if $cd.DependsOn }}
+        dependsOn: [
+            {{- range $cd.DependsOn }}
+            {{- if $i }}
+            ,
+            {{- end }}
+            {
+                containerName: "{{ .ContainerName }}",
+                condition: "{{ .Condition }}",
+            }
+            {{- end }}
+        ],
+        {{- end }}
+        {{- if $cd.EntryPoint }}
+        entryPoint: {{ $cd.EntryPoint }},
+        {{- end }}
+        essential: {{ $cd.Essential }},
+        {{- if $cd.Image }}
+        image: {{ getVar $cd.Image }}.imageName,
+        {{- end }}
+        {{- if $cd.LogConfiguration }}
+        logConfiguration: {
+            logDriver: "{{ $cd.LogConfiguration.LogDriver }}",
+            options: {
+                {{- range $key, $value := $cd.LogConfiguration.Options }}
+                "{{ $key }}": {{ modelCase $value }},
+                {{- end }}
+            },
+            {{- if $cd.LogConfiguration.SecretOptions }}}
+            secretOptions: [
+                {{- range $key, $value := $cd.LogConfiguration.SecretOptions }}
+                {{- if $key }}
+                ,
+                {{- end }}
+                {
+                    name: "{{ $value.Name }}",
+                    valueFrom: "{{ $value.ValueFrom }}",
+                }
+                {{- end }}
+            ]
+            {{- end }}
+        },
+        {{- end }}
+        {{- if $cd.Memory }}
+        memory: {{ $cd.Memory }},
+        {{- end }}
+        {{- if $cd.MountPoints }}
+        mountPoints: [
+            {{- range $key, $value := $cd.MountPoints.M }}
```

Comment:
Things like this make `HashedSet` seem weird. How would someone else writing iac template know to do this? Nothing actionable right now, just something to think about for later.

Reply:
yeah i get that, we might need a better approach